From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: John Galt <johngaltfirstrun@gmail.com>
Date: Fri, 5 Jun 2026 12:52:10 -0400
Subject: [PATCH 41/42] surfaceflinger: specifically no vote status bar

Stops clock updates, battery or signal changes, or network monitor (if
configured to be shown in status bar) from voting at all.
---
 services/surfaceflinger/Layer.h                    | 2 ++
 services/surfaceflinger/Scheduler/LayerHistory.cpp | 4 ++++
 2 files changed, 6 insertions(+)

diff --git a/services/surfaceflinger/Layer.h b/services/surfaceflinger/Layer.h
index 90b5538..6104d18 100644
--- a/services/surfaceflinger/Layer.h
+++ b/services/surfaceflinger/Layer.h
@@ -367,6 +367,8 @@ public:
 
     virtual uid_t getOwnerUid() const { return mOwnerUid; }
 
+    gui::WindowInfo::Type getWindowType() const { return mWindowType; }
+
     // Used to check if mUsedVsyncIdForRefreshRateSelection should be expired when it stop updating.
     nsecs_t mMaxTimeForUseVsyncId = 0;
     // True when DrawState.useVsyncIdForRefreshRateSelection previously set to true during updating
diff --git a/services/surfaceflinger/Scheduler/LayerHistory.cpp b/services/surfaceflinger/Scheduler/LayerHistory.cpp
index 205bbcf..5407c1a 100644
--- a/services/surfaceflinger/Scheduler/LayerHistory.cpp
+++ b/services/surfaceflinger/Scheduler/LayerHistory.cpp
@@ -355,6 +355,10 @@ void LayerHistory::partitionLayers(nsecs_t now) {
                 info->resetLayerVote();
             }
 
+            if (layerUnsafe->getWindowType() == gui::WindowInfo::Type::STATUS_BAR) {
+                info->setLayerVote({LayerVoteType::NoVote, Fps()});
+            }
+
             it++;
         } else {
             if (CC_UNLIKELY(mTraceEnabled)) {
